### ISSUE TITLE
How to get the ground truth?

### ISSUE THREAD

**Issue by feature2** [fh2019ustc/DocTr-Plus#6]
The loss function is defined as the L1 distance between the predicted warping flow f_b and its given ground truth f_{gt}. However, how is the ground truth obtained? In the link [UDIR test set](https://drive.google.com/drive/folders/15rknyt7XE2k6jrxaTc_n5dzXIdCukJLh?usp=share_link), there are only original and scanned images, but there is no ground truth matrix of pixel-wise corresponding between the two images.

**Comment by fh2019ustc** [comment]
Hi, we train our network using the Doc3D dataset and then evaluate it using the UDIR dataset.
The Doc3D dataset contains the ground truth warping flow between the distorted and scanned images.

**Comment by feature2** [comment]
> Hi, we train our network using the Doc3D dataset and then evaluate it using the UDIR dataset. The Doc3D dataset contains the ground truth warping flow between the distorted and scanned images.

Thanks! I have some other questions about Doc3D dataset.
1. Dose 'bm' files store backward warping flow and 'uv' files store forward waring flow?
2. The size of image and bm is 448*448. If I want to resize to 512*512, I can use cv2.resize(image) to resize image, but how can i   
get the resized warping flow?
Thank you in advance for your time and consideration. If you could kindly provide me with answers to these questions, I would greatly appreciate it.

**Comment by zhaolitc** [comment]
@Moonflynn  Hi，have you got how to resize bm file to new resolution？I have tried to use cv2.resize to process bm file just like RGB image，but when I trained the DocTr,  the result is not good. Maybe we should multiply each element value in bm by a resolution-dependent scaling factor?

**Comment by CV-deeplearning** [comment]
> > Hi, we train our network using the Doc3D dataset and then evaluate it using the UDIR dataset. The Doc3D dataset contains the ground truth warping flow between the distorted and scanned images.
> 
> Thanks! I have some other questions about Doc3D dataset.
> 
> 1. Dose 'bm' files store backward warping flow and 'uv' files store forward warping flow?
> 2. The size of image and bm is 448*448. If I want to resize to 512*512, I can use cv2.resize(image) to resize image, but how can i get the resized warping flow?
> 
> Thank you in advance for your time and consideration. If you could kindly provide me with answers to these questions, I would greatly appreciate it.

我实现了flow的resize

**Comment by zhaolitc** [comment]
@CV-deeplearning 你好，可以分享一下你的实现原理或者代码吗？
